reference/api/settings.md

```diff
@@ -1,201 +1,1407 @@
+
+#### Example
+
+<CodeSamples id="update_stop_words_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+  "taskUid": 1,
+  "indexUid": "movies",
+  "status": "enqueued",
+  "type": "settingsUpdate",
+  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+### Reset stop words
+
+<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/stop-words" />
+
+Reset the list of stop words of an index to its default value.
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="reset_stop_words_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+  "taskUid": 1,
+  "indexUid": "movies",
+  "status": "enqueued",
+  "type": "settingsUpdate",
+  "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+## Synonyms
+
+The `synonyms` object contains words and their respective synonyms. A synonym in Meilisearch is considered equal to its associated word for the purposes of calculating search results.
+
+To learn more about synonyms, refer to our [dedicated guide](/learn/configuration/synonyms.md).
+
+### Get synonyms
+
+<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/synonyms"/>
+
+Get the list of synonyms of an index.
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="get_synonyms_1" />
+
+##### Response: `200 OK`
+
+```json
+{
+  "wolverine": [
+    "xmen",
+    "logan"
+  ],
+  "logan": [
+    "wolverine",
+    "xmen"
+  ],
+  "wow": [
+    "world of warcraft"
+  ]
+}
+```
+
+### Update synonyms
+
+<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/synonyms"/>
+
+Update the list of synonyms of an index. Synonyms are [normalized](/learn/configuration/synonyms.md#normalization).
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Body
+
+An object that contains all synonyms and their associated words.
```

Comment:
If possible, I think it would be helpful to add a brief example/explanation of the form that the synonym object takes, so that the user can understand how to structure the object without clicking the below link. On the other hand, there is already a helpful example below so this may be unnecessary labor.

I'm genuinely not sure if it's necessary so I leave the decision to you Maryam, as the person who would actually have to write it 😄